ingredient/ingredient-postgresql/src/plugin.ts

```diff
@@ -1,28 +1,145 @@
+    _functions: PostgreSQLDBUtils; 
+    private _access: string;
+    private _armTemplate: any;
 
     public async Execute(): Promise<void> {
         try {
-            let util = IngredientManager.getIngredientFunction("coreutils", this._ctx)
-            this._logger.log('PostgreSQL Plugin Logging: ' + this._ingredient.properties.parameters)
+            var params = await this._helper.BakeParamsToARMParamsAsync(this._name, this._ingredient.properties.parameters)
+            this.validateBakeParams(params);
+
+        } catch (error){
+            this._logger.error('Bake validation failed: ' + error)
+            throw error;
+        }
+
+        var vnetData = await this.getVnetData(params);
+        params.vnetData = vnetData;
 
-            let params = await this._helper.BakeParamsToARMParamsAsync(this._name, this._ingredient.properties.parameters)
+        if (this._access == "private")
+        {
+            // The Private ARM template includes a few Microsoft.Resources/deployments which should be uniquely named
+            let timestamp = new Date().toISOString().replace(/[^a-zA-Z0-9]/g, "");
+            params.virtualNetworkDeploymentName = {value: `virtualNetwork_${timestamp}`};
+            params.virtualNetworkLinkDeploymentName = {value: `virtualNetworkLink_${timestamp}`};
+            params.privateDnsZoneDeploymentName = {value: `privateDnsZone_${timestamp}`};
+            
+            // Hard coding this for security
+            params.publicNetworkAccess = {value: 'Disabled'};
+        }
+
+        if (!params.firewallRules)
+        {
+            params.firewallRules = {value: {rules: [] }};
+        }
 
-            await this._helper.DeployTemplate(this._name, ARMTemplate, params, await util.resource_group())
+        this.trimParametersForARM(params);
+
+        try {
+            let util = IngredientManager.getIngredientFunction("coreutils", this._ctx);
+            this._logger.log('PostgreSQL Plugin Logging: ' + this._ingredient.properties.parameters)
+
+            await this._helper.DeployTemplate(this._name, this._armTemplate, params, await util.resource_group())
 
         } catch(error){
-            this._logger.error('deployment failed: ' + error)
+            this._logger.error('Deployment failed: ' + error)
             throw error
         }
     }
-}
\ No newline at end of file
+
+    private async getVnetData(params: any): Promise<VnetData> {
+        //var data: any = {}};
+        let util = IngredientManager.getIngredientFunction("coreutils", this._ctx);
+
+        let vNet: VirtualNetwork = await this._functions.get_vnet(params.virtualNetworkResourceGroup.value, params.virtualNetworkName.value)
+        let subnetPropertiesGet: Subnet = await this._functions.get_subnet(params.virtualNetworkResourceGroup.value, params.virtualNetworkName.value, params.subnetName.value)
+        let privateDnsZoneName = this._functions.create_resource_uri(this._access); // todo investigate public equivalent.
```

Comment:
right on. after doing said investigation it looks like there is no action needed, since the public ARM template includes a default value for privateDnsZoneArmResourceId and we don't have any foreseeable need to implement a custom dns zone for public deployments.